# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/events/ContentEventHandler.cpp
# Commit: 5f90ad3d8781
# Full Hash: 5f90ad3d87818a0810f74bde3bfc71d485eea801
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-02-08 17:32:27
# Regressor Bug: 1746104
# File Overlap Count: 1
# Description:
#   Bug 1746104 - part 5-1: Get rid of `WidgetQueryContentEvent::Reply::mHasSelection` r=m_kato
#   
#   It's intended to indicate whether the selection is collapsed or not, but it can
#   be referred by other members, there is no reasonable user and the name makes
#   developers confused.
# ==============================================================================

diff -r dc81d96d0adb -r 5f90ad3d8781 dom/events/ContentEventHandler.cpp
--- a/dom/events/ContentEventHandler.cpp	Mon Feb 07 22:33:37 2022 +0000
+++ b/dom/events/ContentEventHandler.cpp	Mon Feb 07 22:33:37 2022 +0000
@@ -423,8 +423,6 @@
 
   aEvent->mReply->mContentsRoot = mRootContent.get();
 
-  aEvent->mReply->mHasSelection = !mSelection->IsCollapsed();
-
   nsRect r;
   nsIFrame* frame = nsCaret::GetGeometry(mSelection, &r);
   if (!frame) {
@@ -1321,7 +1319,7 @@
   if (!mFirstSelectedRawRange.IsPositioned()) {
     MOZ_ASSERT(aEvent->mInput.mSelectionType != SelectionType::eNormal);
     MOZ_ASSERT(aEvent->mReply->mOffsetAndData.isNothing());
-    MOZ_ASSERT(!aEvent->mReply->mHasSelection);
+    MOZ_ASSERT_IF(mSelection, !mSelection->RangeCount());
     // This is special case that `mReply` is emplaced, but mOffsetAndData is
     // not emplaced but treated as succeeded because of no selection ranges
     // is a usual case.
@@ -2546,7 +2544,7 @@
 
   MOZ_ASSERT(aEvent->mReply.isSome());
 
-  if (!aEvent->mReply->mHasSelection) {
+  if (mSelection->IsCollapsed()) {
     MOZ_ASSERT(!aEvent->mReply->mTransferable);
     return NS_OK;
   }